counsel.el

```diff
@@ -5048,6 +5048,12 @@ Local bindings (`counsel-mode-map'):
     (when (fboundp 'advice-remove)
       (advice-remove #'describe-bindings #'counsel-descbinds))))
 
+(dolist (cmd ivy-highlight-grep-commands)
+  (add-to-list 'ivy-initial-inputs-alist
+               (cons cmd 'ivy-word-at-point-dwim)
+               nil
+               (lambda (cell1 cell2) (eq (car cell1) (car cell2)))))
+
```

Comment:
Further problem is that if `ivy` is loaded before `counsel`, then `ivy-highlight-grep-commands` is simply `nil` and that modification of `initial-inputs-alist` does nothing.  Fragile.

Reply:
> Further problem is that if `ivy` is loaded before `counsel`, then `ivy-highlight-grep-commands` is simply `nil` and that modification of `initial-inputs-alist` does nothing. Fragile.

I take back the last comment as I overlooked that you added this code to `counsel` instead of `ivy`.  Forceful default initialization of `ivy-initial-inputs-alist` still applies though.